plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/StandardProjectWizard.java

```diff
@@ -58,10 +57,26 @@ public boolean performFinish() {
       status = new Status(Status.ERROR, "todo plugin ID", errorCode, ex.getMessage(), null);
     }
     
-    // todo if fail, call setErrorMessage()
+    showErrorMessageIfNecessary(status);
+    
     return status.isOK();
   }
 
+  
+  // visible for testing
+  void showErrorMessageIfNecessary(IStatus status) {
+    // if fail, display error message in wizard
+    if (status == Status.CANCEL_STATUS) {
+      page.setErrorMessage("User canceled project creation");
+    } else if (!status.isOK()) {
+      String message = "Failed to create project";
+      if (status.getMessage() != null && !status.getMessage().isEmpty()) {
+        message += ": " + status.getMessage();
+      }
+      page.setErrorMessage(message);
```

Comment:
JDT's `org.eclipse.jdt.internal.ui.wizards.JavaProjectWizard` opens an error dialog on exception during its performFinished().  I'd follow their lead.  The `WizardPage#setErrorMessage()` is more meant for validation errors.

My real problem with using `page.setErrorMessage()` is that if we have a different page, the setErrorMessage() is ignored with no warning.